src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -14,13 +18,62 @@
   "v3")
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(defn class-info-from-jruby->class-info-for-json
+  "Convert a class info map received from the jruby service into an
+  appropriate map for use in serializing an environment_classes response to
+  JSON"
+  [info-from-jruby environment]
```

Comment:
I added some schema for this in 2dd5530.  I only had it go fairly top-level - just looking to see that what we get back from the JRuby layer is a `Map` and that what we translate for parsing to JSON has the intended structure down to a per-file level - but not individual classes within those files.  Hope you think that's reasonable for now.